lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -410,42 +413,306 @@ class ConvertAten_IndexPutImplOp
           op, "Indices list size should not be greater than the input rank.");
 
     // TODO: Add support for cases with indices list size not equal to 1.
-    if (indicesList.size() != 1)
-      return rewriter.notifyMatchFailure(
-          op, "Unimplemented: Indices list size != 1");
-    Value indexTensor = indicesList[0];
-
-    if (indexTensor.getType().isa<Torch::NoneType>())
-      return rewriter.notifyMatchFailure(op, "Index tensor must not be None.");
-
-    // Creating a tm_tensor.scatter op with the following mapping:
-    // 1.) Index tensor from the `indicesList` maps to the indices in scatter
-    // op. Index tensor is expanded from 1-d to 2-d, and its element type is set
-    // to i32 as required for the scatter op.
-    // 2.) `values` is mapped to `updates` in scatter op.
-    // 3.) `input` is mapped to `original` in scatter op.
-    std::optional<unsigned> indexTensorRank = getTensorRank(indexTensor);
-    if (!indexTensorRank || *indexTensorRank != 1)
-      return rewriter.notifyMatchFailure(
-          op, "unimplemented: index tensor with rank != 1 is not supported");
-    auto indexTensorType = indexTensor.getType().cast<BaseTensorType>();
-    int64_t indexTensorSize = indexTensorType.getSizes()[0];
-    SmallVector<int64_t> expandedIndexTensorSizes{indexTensorSize, 1};
-    ValueTensorType expandedIndexTensorType =
-        ValueTensorType::get(context, llvm::ArrayRef(expandedIndexTensorSizes),
-                             indexTensorType.getDtype());
+    Value torchCstZero = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(0));
     Value torchCstOne = rewriter.create<Torch::ConstantIntOp>(
         loc, rewriter.getI64IntegerAttr(1));
-    Value expandedIndexTensor = rewriter.create<AtenUnsqueezeOp>(
-        loc, expandedIndexTensorType, indexTensor, torchCstOne);
+    Value torchCstNone = rewriter.create<Torch::ConstantNoneOp>(loc);
+    Value indexTensor, flattenedValuesTensorLastDim;
+    BaseTensorType indexTensorType;
+    bool needToFlattenValuesTensor = false;
+    SmallVector<int64_t> flattenedValuesTensorSize;
+    if (indicesList.size() == 1) {
```

Comment:
Why do we need to have a special case for 1 index tensor? I would've expected the logic for 2 index tensors to work for 1. Moreover, using the code for 2 index tensors would allow removing the restrictions on `indexTensorRank`